File: clients/android/NewsBlur/src/com/newsblur/widget/WidgetRemoteViewsService.java
package com.newsblur.widget;

import android.content.Intent;
import android.widget.RemoteViewsService;

import com.newsblur.util.Log;

public class WidgetRemoteViewsService extends RemoteViewsService {

    private static String TAG = "WidgetRemoteViewsFactory";

    @Override
    public RemoteViewsService.RemoteViewsFactory onGetViewFactory(Intent intent) {
        Log.d(TAG, "onGetViewFactory");
        return new WidgetRemoteViewsFactory(this.getApplicationContext(), intent);
    }
}

File: clients/android/NewsBlur/app/src/main/java/com/newsblur/widget/WidgetRemoteViewsService.kt
package com.newsblur.widget

import android.content.Intent
import android.widget.RemoteViewsService
import com.newsblur.util.Log

class WidgetRemoteViewsService : RemoteViewsService() {

    override fun onGetViewFactory(intent: Intent): RemoteViewsFactory =
            WidgetRemoteViewsFactory(this.applicationContext, intent).also {
                Log.d("WidgetRemoteViewsFactory", "onGetViewFactory")
            }
}